### PR TITLE
Run discovery more often

### PR DESCRIPTION
**Pull request by grunskis**
Run device discovery every 15 seconds during onboarding and then stop it after we're done.

I'm excluding `network_discovery.py ` from test coverage check. It's mostly copy/paste from netdisco. IMO we should rewrite it to make it simpler and then test it, but for now it works as it is.

### CODE REVIEW

File: senic_hub/backend/device_discovery.py [hunk 1]
```diff
@@ -77,13 +77,10 @@ def discover(discovery_class=NetworkDiscovery):
 
     devices = []
 
-    netdisco = discovery_class()
+    netdisco = discovery_class(limit_discovery=SUPPORTED_DEVICES)
```
Comment (larsblumberg) [inline senic_hub/backend/device_discovery.py]: It's not clear to me that `limit_discovery` takes a list of supported devices. It sounds like boolean variable to me. I suggest `device_filter`.

File: senic_hub/backend/network_discovery.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+
+class NetworkDiscovery(object):
+    """
+    Based on netdisco.discovery.NetworkDiscovery. This class only does
+    discovery using SSDP & nUPNP which covers all the devices we want
+    to support at the moment.
+
+    """
+    def __init__(self, limit_discovery=[]):
```
Comment (larsblumberg) [inline senic_hub/backend/network_discovery.py]: What about creating a PR to netdisco that introduces this optional parameter?

Comment (larsblumberg) [inline senic_hub/backend/network_discovery.py]: Default value should be `None`, not an empty array. Empty array should not include any device.

File: senic_hub/backend/device_discovery.py (hunk 1 above)
Comment (grunskis) [inline senic_hub/backend/device_discovery.py]: It's not really a filter. It's a list of devices to search for. Would `discover_devices` or `device_whitelist` work better as a name for it?

Comment (larsblumberg) [inline senic_hub/backend/device_discovery.py]: `whitelist` sounds good!

File: senic_hub/backend/network_discovery.py (hunk 2 above)
Comment (larsblumberg) [inline senic_hub/backend/network_discovery.py]: (updated comment, was hard to understand)

Comment (grunskis) [inline senic_hub/backend/network_discovery.py]: Yep, that what the intention. I'll remove the default value, which would make it more clear.

File: senic_hub/backend/device_discovery.py (hunk 1 above)
Comment (grunskis) [inline senic_hub/backend/device_discovery.py]: Addressed in f631c15

**Pull request merged**